nrfx_config_nrf5340_application.h

```diff
@@ -101,13 +95,27 @@
 #define NRF_UARTE1       NRF_PERIPH(NRF_UARTE1)
 #define NRF_UARTE2       NRF_PERIPH(NRF_UARTE2)
 #define NRF_UARTE3       NRF_PERIPH(NRF_UARTE3)
-#define NRF_UICR         NRF_PERIPH(NRF_UICR)
 #define NRF_USBD         NRF_PERIPH(NRF_USBD)
 #define NRF_USBREGULATOR NRF_PERIPH(NRF_USBREGULATOR)
 #define NRF_VMC          NRF_PERIPH(NRF_VMC)
 #define NRF_WDT0         NRF_PERIPH(NRF_WDT0)
 #define NRF_WDT1         NRF_PERIPH(NRF_WDT1)
 
+/*
+ * The following section provides the name translation for peripherals with
+ * only one type of access available. For these peripherals, you cannot choose
+ * between secure and non-secure mapping.
+ */
+#define NRF_CACHE      NRF_CACHE_S
```

Comment:
Could you add an !NRF_TRUSTZONE_NONSECURE guard here, as well?

Reply:
I am not entirely convinced that this will be good. This way we'll make it impossible to compile code like the mentioned nrfx_nfct driver in its current form. But we still won't prevent users from misusing `*_S` symbols. One can still use one of those in a non-secure image, compile code successfully, and encounter the SecureFault. What is the gain then?

Reply:
Users should not be allowed to use the _S or _NS_ defines, that's the whole point, really; those defines are supposed to remain hidden from the user.  [Btw, nothing prevents a used to generate a secure fault (e.g. they can define a ptr = 0x50000000 and de-reference it directly :p) but we should try be of some assistance here.]

So the principles should be:
- users are **only** allowed to use NRF_"PERIPH" defines in software; that's why we do the effort and define them here. [An exception could be a software driver that is by-definition a Secure-only driver, e.g. cryptocell].
- user shall be allowed to use **any** NRF_"PERIPH" if one is defined, and that should be always a safe thing to do.

The first bullet should be enforced by definition, as an nrfx coding principle.

Your patch addresses the second bullet, by deferring from defining NRF_"PERIPH" for Non-Secure builds, if the peripheral is Secure-only. But for this to be complete, you need to  guard the explicit NRF_"PERIPH" defines for _S only peripherals. Same for peripherals that are NS-only; the defines should be guarded accordingly.

In other words: if compilation passes, the driver should _not_ be able to trigger secure faults. If we don't ensure this, then Zephyr won't qualify as a "safe" Non-Secure firmware (from ARM's PSA perspective).

>This way we'll make it impossible to compile code like the mentioned nrfx_nfct driver in its current form.

Correct, but i think these problems are easily solved by guarding the driver functions with the !NONSECURE macro guard, and document in the header that these are secure-only functions.

Reply:
Okay, that makes sense. I think I got convinced. 😉
I modified the PR accordingly.